src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -972,7 +975,7 @@ public void InvalidateCacheEntry(ActivationAddress activationAddress, bool inval
             // for multi-cluster registration, the local directory may cache remote activations
             // and we need to remove them here, on the fast path, to avoid forwarding the message
             // to the wrong destination again
-            if (invalidateDirectoryAlso && CalculateTargetSilo(grainId).Equals(MyAddress))
+            if (invalidateDirectoryAlso && CalculateGrainDirectoryPartition(grainId).Equals(MyAddress))
```

Comment:
If `CalculateGrainDirectoryPartition` returns `null`, this will cause a NRE. Wouldn't it be safer to flip the check to
```csharp
MyAddress.Equals(CalculateGrainDirectoryPartition(grainId))
``` 
instead.